src/client/Microsoft.Identity.Client/ManagedIdentity/V2/ImdsV2ManagedIdentitySource.cs

```diff
@@ -214,7 +214,7 @@ private async Task<CertificateRequestResponse> ExecuteCertificateRequestAsync(
                 { "x-ms-client-request-id", _requestContext.CorrelationId.ToString() }
             };
             
-            var payload = new PemPayload { pem = pem };
+            var payload = new PemPayload { pem = csrPem };
             var body = JsonHelper.SerializeToJson(payload);
```

Comment:
Do you really need to create a new object just to serialize it in a specific way? Would it be simpler to just use a string template here instaed, e.g. `string body = @"{pemPayload=<payload>}` ? (you'll need to escape { correctly)